Sources/NIO/EventLoop.swift

```diff
@@ -291,6 +295,74 @@ extension TimeAmount: Comparable {
     }
 }
 
+extension TimeAmount {
+    public static func + (lhs: TimeAmount, rhs: TimeAmount) -> TimeAmount {
+        return TimeAmount(lhs.nanoseconds + rhs.nanoseconds)
+    }
+
+    public static func - (lhs: TimeAmount, rhs: TimeAmount) -> TimeAmount {
+        return TimeAmount(lhs.nanoseconds - rhs.nanoseconds)
+    }
+
+    public static func * (lhs: TimeAmount.Value, rhs: TimeAmount) -> TimeAmount {
+        return TimeAmount(lhs * rhs.nanoseconds)
+    }
+
+    public static func * (lhs: TimeAmount, rhs: TimeAmount.Value) -> TimeAmount {
+        return TimeAmount(lhs.nanoseconds * rhs)
+    }
```

Comment:
The bottom two implementations are written against `Int64`. Should we instead write them to be more generic, e.g. over `BinaryInteger`? I'm not sure that we should, but I *definitely* think we should also write an implementation that uses `Int`.

Reply:
I added an `Int` version